attestor/root.go

```diff
@@ -0,0 +1,305 @@
+		doCheck, doSign = true, true
+		modeExample = `	./scorecard-attestor \
+	-mode=check-and-sign \
+	-image=gcr.io/my-image-repo/image-1@sha256:123 \
+	-policy=policy.yaml \
+	-note_name=projects/$NOTE_PROJECT/NOTES/$NOTE_ID \
+	-kms_key_name=projects/$KMS_PROJECT/locations/$KMS_KEYLOCATION/keyRings/$KMS_KEYRING/cryptoKeys/$KMS_KEYNAME/cryptoKeyVersions/$KMS_KEYVERSION \
+	-kms_digest_alg=SHA512`
+	case BypassAndSign:
+		addSignFlags(modeFlags)
+		doSign = true
+		modeExample = `	./scorecard-attestor \
+	-mode=bypass-and-sign \
+	-image=gcr.io/my-image-repo/image-1@sha256:123 \
+	-note_name=projects/$NOTE_PROJECT/NOTES/$NOTE_ID \
+	-kms_key_name=projects/$KMS_PROJECT/locations/$KMS_KEYLOCATION/keyRings/$KMS_KEYRING/cryptoKeys/$KMS_KEYNAME/cryptoKeyVersions/$KMS_KEYVERSION \
+	-kms_digest_alg=SHA512`
+	case CheckOnly:
+		addCheckFlags(modeFlags)
+		doCheck = true
+		modeExample = `	./scorecard-attestor \
+	-mode=check-only \
+	-image=gcr.io/my-image-repo/image-1@sha256:123 \
+	-policy=policy.yaml`
+	default:
+		return false, false, fmt.Errorf("unrecognized mode %s, must be one of check-and-sign|check-only|bypass-and-sign", mode)
+	}
+	flag.Parse()
+	return doCheck, doSign, err
+}
+
+func exitOnBadFlags(mode SignerMode, err string) {
+	fmt.Fprintf(modeFlags.Output(), "Usage of signer's %s mode:\n", mode)
+	modeFlags.PrintDefaults()
+	fmt.Fprintf(modeFlags.Output(), "Example (%s mode):\n %s\n", mode, modeExample)
+	fmt.Fprintf(modeFlags.Output(), "Bad flags for mode %s: %v. \n", mode, err)
+	os.Exit(1)
+}
+
+func main() {
+	flag.Parse()
+	glog.Infof("Signer mode: %s.", mode)
+	ctx := context.Background()
+
+	doCheck, doSign, err := parseSignerMode(SignerMode(mode))
+	if err != nil {
+		glog.Fatalf("Parse mode err %v.", err)
+	}
+
+	// Check image url is non-empty
+	if image == "" {
+		exitOnBadFlags(SignerMode(mode), "image url is empty")
+	}
+
+	// Create a client
+	client, err := containeranalysis.New()
+	if err != nil {
+		glog.Fatalf("Could not initialize the client %v", err)
+	}
+
+	if doCheck {
+		// Read the Binauthz attestation policy
+		if policyPath == "" {
+			exitOnBadFlags(SignerMode(mode), "policy path is empty")
+		}
+		attestationPolicy, err := policy.ParseAttestationPolicyFromFile(policyPath)
+		if err != nil {
+			glog.Fatalf("Fail to load scorecard attestation policy: %v", err)
+		}
+
+		if repoURL == "" {
+			buildRepo := os.Getenv("REPO_NAME")
+			if buildRepo == "" {
+				exitOnBadFlags(SignerMode(mode), "repoURL not specified")
+			}
+			repoURL = buildRepo
+			glog.Infof("Found repo URL %s Cloud Build environment", repoURL)
+		} else {
+			glog.Infof("Running scorecard on %s", repoURL)
+		}
+
+		if commitSHA == "" {
+			buildSHA := os.Getenv("COMMIT_SHA")
+			if buildSHA == "" {
+				glog.Infof("commit not specified, running on HEAD")
+			} else {
+				commitSHA = buildSHA
+				glog.Infof("Found revision %s Cloud Build environment", commitSHA)
+			}
+		}
+
+		logger := sclog.NewLogger(sclog.DefaultLevel)
```

Comment:
can we use a file as input instead of running scorecard?

Reply:
Yes - b/c of the way that the evaluation is structured, it could work with an implementation where we just read in raw results as JSON. But currently I think that running scorecard inside of the attestor is the easiest way in a Cloud Build pipeline or for other use cases since it leads to fewer steps.

Reply:
so long as we can support it (to use the attestor on REST API results), fine with me.
I get the point about "easier", but it's also less flexible in a sense. I'm not sure 1 step vs 2 steps is that much of a convenience in practice. Not worth blocking this PR for this, though :)

Reply:
Yep, not opposed in general.